rust/pkg/cardano_serialization_lib.js.flow

```diff
@@ -2529,6 +2529,57 @@ declare export class MetadataMap {
    */
   keys(): MetadataList;
 }
+/**
+ */
+declare export class MinOutputAdaCalculator {
+  free(): void;
+
+  /**
+   * @param {TransactionOutput} output
+   * @param {BigNum} coins_per_utxo_word
+   * @returns {MinOutputAdaCalculator}
+   */
+  static new(
+    output: TransactionOutput,
+    coins_per_utxo_word: BigNum
+  ): MinOutputAdaCalculator;
+
+  /**
+   * @param {BigNum} coins_per_utxo_word
+   * @returns {MinOutputAdaCalculator}
+   */
+  static new_empty(coins_per_utxo_word: BigNum): MinOutputAdaCalculator;
+
+  /**
+   * @param {Address} address
+   */
+  set_address(address: Address): void;
+
+  /**
+   * @param {PlutusData} data
+   */
+  set_data(data: PlutusData): void;
+
+  /**
+   * @param {DataHash} data_hash
+   */
+  set_data_hash(data_hash: DataHash): void;
+
+  /**
+   * @param {Value} amount
+   */
+  set_amount(amount: Value): void;
+
+  /**
+   * @param {ScriptRef} script_ref
+   */
+  set_script_ref(script_ref: ScriptRef): void;
+
+  /**
+   * @returns {BigNum}
+   */
+  calculate_ada(): BigNum;
+}
```

Comment:
@lisicky , not sure if there's a good reason to have a separate structure and type for the calculator, it has separate setters to accept fields like datum, address, value, ref-script, etc. , but there's already exactly a type for all these fields together - `TransactionOutput`. Think we can just have a function with the `TransactionOutput` as an argument and that's it, otherwise with separate setters it becomes less obvious that you need ALL these fields, to calculate the required value correctly

Reply:
@vsubhuman I agree with you. But main reason for creation the structure was that we have different ways for min ada calculation such as min_pure_ada and min_ada_required and I think it's better to have a one point of min ada calculation for a TransactionOutput to reduce numbers of that functions. We can left the structure for internal usage and create only one function with only TransactionOutput and cost of words/bytes. But if you think that the structure is redundant and useless we can discuss in the slack, because there are some discussable moment in the change estimation function.